### PR TITLE
Replace Disposable[] with DisposableStore

### PR DESCRIPTION
**Pull request by solomatov**
See https://github.com/microsoft/vscode/issues/74250 for details.

### CODE REVIEW

**Comment by solomatov** [comment]
If it's ok, I can do more of it. There aren't that many usages of Disposable[] as a DisposableStore.

File: src/vs/base/browser/ui/list/listView.ts [hunk 1]
```diff
@@ -188,7 +188,7 @@ export class ListView<T> implements ISpliceable<T>, IDisposable {
 	private currentDragFeedbackDisposable: IDisposable = Disposable.None;
 	private onDragLeaveTimeout: IDisposable = Disposable.None;
 
-	private disposables: IDisposable[];
+	private disposables: DisposableStore;
```
Comment (mjbvz) [inline src/vs/base/browser/ui/list/listView.ts]: Please make this property `readonly`. This helps prevent the disposables registered on a `DisposableStore` from being lost with a re-assignment

Comment (solomatov) [reply src/vs/base/browser/ui/list/listView.ts]: Thanks. Fixed.

**Comment by solomatov** [comment]
And what do you think about more of such contributions? I want to learn the codebase, and such changes are a good way to learn about it.

File: src/vs/base/browser/ui/list/listView.ts [hunk 2]
```diff
@@ -214,7 +214,7 @@ export class ListView<T> implements ISpliceable<T>, IDisposable {
 			this.renderers.set(renderer.templateId, renderer);
 		}
 
-		this.cache = new RowCache(this.renderers);
+		this.cache = this.disposables.add(new RowCache(this.renderers));
```
Comment (mjbvz) [inline src/vs/base/browser/ui/list/listView.ts]: Make sure to call `this.disposables.clear()` before adding the new entries

Comment (mjbvz) [reply src/vs/base/browser/ui/list/listView.ts]: Actually never mind. I see this is in the constructor so it's not required

**Comment by mjbvz** [comment]
Thanks. We would take other PRs like this. Please just keep each PR small and isolated so they are easy to review

**Pull request merged**